### PR TITLE
Upgrade to TypeScript 5.1, update TSConfig & eslint-config-seek

### PR DESCRIPTION
**Pull request by mrm007**
- Upgrade to TypeScript 5.1
- Update `tsconfig.json` options to match the latest version of the TypeScript compiler.
- Update dependency `eslint-config-seek` which reverts [the autofix for a Cypress rule](https://github.com/seek-oss/eslint-config-seek/releases/tag/v11.2.1)
- Remove `import React` from template and fixtures

### CODE REVIEW

File: packages/sku/config/typescript/tsconfig.js [hunk 1]
```diff
@@ -24,14 +24,23 @@ module.exports = () => {
...
+      // misc
       strict: true,
-      jsx: 'preserve',
-      lib: ['dom', 'es2019'],
-      target: 'es5',
+      forceConsistentCasingInFileNames: true,
+      noUnusedLocals: true,
+      jsx: 'react-jsx',
+      lib: ['dom', 'es2022'],
+      target: 'es2022',
```
Comment (mrm007) [inline packages/sku/config/typescript/tsconfig.js:43]: Babel takes care of this via `babel-preset-env` + `browserslist-config-seek`

File: packages/sku/config/typescript/tsconfig.js [hunk 2]
```diff
@@ -24,14 +24,22 @@ module.exports = () => {
...
+      // When dependencies have `type: module` TypeScript doesn't know how to resolve them,
+      // so we need to set this to `bundler` or `node`
+      moduleResolution: 'bundler',
+
+      // resolution-related
       allowSyntheticDefaultImports: true,
+      esModuleInterop: true,
+      isolatedModules: true,
       resolveJsonModule: true,
-      noUnusedLocals: true,
```
Comment (mrm007) [inline packages/sku/config/typescript/tsconfig.js:30]: Covered by `eslint-config-seek`
- https://github.com/seek-oss/eslint-config-seek/blob/ad1accdaa0ec37e06ff1dc75970a4bdf92ee74f2/base.js#L55-L58
- https://github.com/seek-oss/eslint-config-seek/blob/ad1accdaa0ec37e06ff1dc75970a4bdf92ee74f2/base.js#L131-L134

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -14202,6 +14440,45 @@ packages:
       - esbuild
       - uglify-js
 
+  /webpack@5.88.0:
```
Comment (askoufis) [inline pnpm-lock.yaml:14443]: Where did this come from?

File: packages/sku/config/typescript/tsconfig.js (hunk 1 above)
Comment (askoufis) [reply packages/sku/config/typescript/tsconfig.js:42]: Doesn't `ESNext` have features that are still in proposal stages? I guess it's fine, but it also would require us to keep babel up to date more frequently.

File: pnpm-lock.yaml (hunk 3 above)
Comment (mrm007) [reply pnpm-lock.yaml:14443]: Looks like it came via fixtures 

https://github.com/seek-oss/sku/blob/8d11309848dc183184099578af4b95cd5a99edb1/pnpm-lock.yaml#L344-L354

Comment (mrm007) [reply pnpm-lock.yaml]: I ran a `pnpm dedupe` and it removed some of them, but there's still two of them.

File: packages/sku/config/typescript/tsconfig.js (hunk 1 above)
Comment (mrm007) [reply packages/sku/config/typescript/tsconfig.js:42]: I'll leave it, then. This PR has ballooned out of proportion.

File: pnpm-lock.yaml (hunk 3 above)
Comment (mrm007) [reply pnpm-lock.yaml]: [`e9983f9` (#844)](https://github.com/seek-oss/sku/pull/844/commits/e9983f9e0cc6be17cff6065a07489f2f103acfa5)

File: packages/sku/config/eslint/eslintConfig.js [hunk 4]
```diff
@@ -6,6 +6,11 @@ const baseConfig = {
   extends: orderImports
     ? [coreConfig, require.resolve('./importOrderConfig')]
     : coreConfig,
+  rules: {
+    // Temporarily disable this rule until eslint-config-seek is updated.
+    // This goes hand-in-hand with `compilerOptions.jsx: 'react-jsx'` in tsconfig.json
+    'react/react-in-jsx-scope': 'off',
```
Comment (mrm007) [inline packages/sku/config/eslint/eslintConfig.js:12]: 1. Release sku
2. Update and release eslint-config-seek
3. Come back here and remove the exception

File: packages/sku/config/typescript/tsconfig.js [hunk 5]
```diff
@@ -24,14 +24,25 @@ module.exports = () => {
       // Fixes https://github.com/cypress-io/cypress/issues/1087
       skipLibCheck: true,
 
-      esModuleInterop: true,
+      // When dependencies have `type: module` TypeScript doesn't know how to resolve them,
+      // so we need to set this to `bundler` or `node`
+      moduleResolution: 'bundler',
+      // This is required by `bundler` and for dynamic `import()`
+      module: 'es2022',
```
Comment (mrm007) [inline packages/sku/config/typescript/tsconfig.js:31]: Confirmed this is still valid with [TypeScript 5.2](https://devblogs.microsoft.com/typescript/announcing-typescript-5-2-beta/#module-and-moduleresolution-must-match-under-recent-node-js-settings)

File: packages/sku/config/typescript/tsconfig.js [hunk 6]
```diff
@@ -24,14 +24,25 @@ module.exports = () => {
...
+
+      // resolution-related
+      allowImportingTsExtensions: true,
```
Comment (mrm007) [inline packages/sku/config/typescript/tsconfig.js:34]: In [ESM mode](https://www.typescriptlang.org/docs/handbook/esm-node.html) extensions are mandatory. I find it weird importing from `.js` files when the file on disk is `.ts`.

**Review by askoufis — APPROVED** [review]

**Pull request merged**